### PR TITLE
2022-12-08 BryanSoltis - v2.3.2 Updates

### PR DESCRIPTION
**Pull request by BryanSoltis**
- Updated Reference/Generate pages to reload configuration data, if empty
- Updated site to exclude Custom Component if no options are entered
- Updated resourcetypes.json regex patterns
- Updated cache reload process to ensure all data is loaded
- Updated Admin Page styling
- Updated comments in API
- Updated Reference page to include delimiter for samples
- Added warning to Generate page for resource types with a lower-level scope
- Updated Nuget packages

Notification:  @jocontr @dominicallen @EdPrice-MSFT

### CODE REVIEW

File: ready/AzNamingTool/Attributes/RateFilter.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                }
+
+                if (minResponseRateFeature != null)
+                {
+                    minResponseRateFeature.MinDataRate = new MinDataRate(bytesPerSecond: 100, gracePeriod: TimeSpan.FromSeconds(10));
+                }
+            }
+            catch (Exception ex)
+            {
+                //Log or Throw
```
Comment (jocontr) [inline ready/AzNamingTool/Attributes/RateFilter.cs:36]: Log to your default sink.

File: ready/AzNamingTool/Helpers/GeneralHelper.cs [hunk 2]
```diff
@@ -196,12 +194,11 @@ public async static Task WriteList<T>(List<T> items)
                 };
 
                 // Update the cache with the latest data
-                SetCacheObject(typeof(T).Name, data);
+                CacheHelper.SetCacheObject(typeof(T).Name, data);
             }
             catch (Exception ex)
             {
                 AdminLogService.PostItem(new AdminLogMessage() { Title = "ERROR", Message = ex.Message });
-                throw;
```
Comment (jocontr) [inline ready/AzNamingTool/Helpers/GeneralHelper.cs:204]: Good removal, compound throws and unwinding the stack are expensive operations.

File: ready/AzNamingTool/Helpers/ValidationHelper.cs [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                    {
+                        valid = true;
+                    }
+                    break;
+            }
+
+            return valid;
+        }
+
+        public static Tuple<bool, string, StringBuilder> ValidateGeneratedName(Models.ResourceType resourceType, string name, string delimiter)
```
Comment (jocontr) [inline ready/AzNamingTool/Helpers/ValidationHelper.cs:80]: Not a fan of tuples, I prefer strong types. Something to consider.

**Review by jocontr — APPROVED** [review]
Looks good; just minor comments.

File: ready/AzNamingTool/Attributes/RateFilter.cs (hunk 1 above)
Comment (BryanSoltis) [reply ready/AzNamingTool/Attributes/RateFilter.cs]: Added logging to Admin log.

**Pull request merged**